### PR TITLE
fix: use static network connectors

### PR DESCRIPTION
**Pull request by zzmp**
Use static connectors for network connections to avoid constant calls to eth_blockNumber

### CODE REVIEW

File: src/hooks/web3/useJsonRpcUrlMap.tsx [hunk 1]
```diff
@@ -30,3 +15,36 @@ export default function useJsonRpcUrlMap(): Record<SupportedChainId, string[]> {
...
+    }
+  }, {} as Record<SupportedChainId, (JsonRpcProvider | string)[]>)
+}
+
+export function toJsonRpcConnectionMap(
+  connectionMap?: JsonRpcConnectionMap
+): Record<SupportedChainId, [JsonRpcProvider]> {
+  return Object.entries(toJsonRpcMap(connectionMap)).reduce((map, [chainId, connections]) => {
+    const connection = connections[0]
+    const provider = JsonRpcProvider.isProvider(connection) ? connection : new StaticJsonRpcProvider(connection)
```
Comment (zzmp) [inline src/hooks/web3/useJsonRpcUrlMap.tsx:37]: This must be provided with a default chainId, or it will immediately send a JSON-RPC request to determine the network.

File: src/hooks/web3/useConnectors.tsx [hunk 2]
```diff
@@ -16,7 +17,17 @@ export interface Connectors {
 
 const ConnectorsContext = createContext<Connectors | null>(null)
 
-export const Provider = ConnectorsContext.Provider
+export function Provider({ value, children }: PropsWithChildren<{ value: Connectors }>) {
```
Comment (vm) [inline src/hooks/web3/useConnectors.tsx:20]: can we rename value to connectors?

File: src/hooks/web3/useJsonRpcUrlMap.tsx [hunk 3]
```diff
@@ -30,3 +15,38 @@ export default function useJsonRpcUrlMap(): Record<SupportedChainId, string[]> {
...
+  return ALL_SUPPORTED_CHAIN_IDS.reduce((map, chainId) => {
+    const value = connectionMap?.[chainId]
+    const connections = (Array.isArray(value) ? value : [value])
+      .filter((value): value is string | JsonRpcProvider => Boolean(value))
+      .concat(...JSON_RPC_FALLBACK_ENDPOINTS[chainId])
+    return {
+      ...map,
+      [chainId]: connections,
+    }
+  }, {} as Record<SupportedChainId, (JsonRpcProvider | string)[]>)
```
Comment (vm) [inline src/hooks/web3/useJsonRpcUrlMap.tsx:29]: is there an alternative to casting here? someone could change the logic and break this in the future

File: src/hooks/web3/useJsonRpcUrlMap.tsx [hunk 4]
```diff
@@ -30,3 +15,38 @@ export default function useJsonRpcUrlMap(): Record<SupportedChainId, string[]> {
...
+}
+
+export function toJsonRpcConnectionMap(
```
Comment (vm) [inline src/hooks/web3/useJsonRpcUrlMap.tsx:32]: should the prop be `jsonRpcMap` instead of needing to re-create for this and `toJsonRpcUrlMap`?

File: e2e/connect.test.tsx [hunk 5]
```diff
@@ -28,31 +28,31 @@ describe('connect', () => {
     })
   }
 
-  function itExpectsWidgetToBeDisabled(renderWidget: () => RenderResult, disabled = true) {
```
Comment (vm) [inline e2e/connect.test.tsx:31]: why the changes here?

Comment (vm) [reply e2e/connect.test.tsx:31]: oh nvm i understand

File: src/hooks/web3/useJsonRpcUrlMap.tsx (hunk 4 above)
Comment (zzmp) [reply src/hooks/web3/useJsonRpcUrlMap.tsx:45]: I don't understand what you're asking.

File: src/components/Widget.tsx [hunk 6]
```diff
@@ -100,7 +100,7 @@ export interface WidgetProps extends BrandingSettings, TransactionEventHandlers
   theme?: Theme
   locale?: SupportedLocale
   provider?: Eip1193Provider | JsonRpcProvider
-  jsonRpcUrlMap?: JsonRpcUrlMap
```
Comment (zzmp) [inline src/components/Widget.tsx:103]: I think this is currently broken / unused :/

**Review by vm — APPROVED** [review]

**Review by vm — APPROVED** [review]
